Project/src/main/java/app/FileUploadController.java

```diff
@@ -58,7 +60,9 @@ public String uploadFile(@RequestParam("file") MultipartFile file) throws IOExce
     @GetMapping("/getfile/{id}")
     public ResponseEntity<byte[]> getFileById(@PathVariable(value = "id")Long id){
         Article article = articleRepository.findById(id).get();
-        if (!article.getUsers().contains(super.getUser())) {
+        User currentUser = super.getUser();
+        // allow editors to download all files
+        if (!article.getUsers().contains(currentUser) && currentUser.getRole() != Role.EDITOR) {
```

Comment:
Submitters also should be able to download a file. @acournoyer1 please confirm but I think reviewers should as well correct? I think the if statement should be this
```suggestion
        if (user == null || !article.getUsers().contains(currentUser)) {
```

Reply:
Both reviewers and submitters should also be able to download but only if they are authorized for that specific article, the AND here is special because editors are not explicitly added to the article.authorizedUsers list. If we did want to add them we would have to query for all editors and add each one, I thought this was a better solution.

Reply:
ah gotchu. kk that makes more sense.